setup.py

```diff
@@ -12,6 +12,7 @@
     'colorama;sys_platform=="win32"',
     "pluggy>=0.12,<1.0",
     'importlib-metadata>=0.12;python_version<"3.8"',
+    'typing_extensions;python_version<"3.5.2"',
```

Comment:
Let's please add a trivial changelog for this:

```rst
Added ``typying_extensions`` as dependency on Python 3.5.1 or lower.
```

This is particularly interesting for alternative package maintainers (such as `conda`, which I maintain) because they need to be aware of the new dependency. 👍